### ISSUE TITLE
Casks with `appcast` and `:no_check`

### ISSUE THREAD

**Issue by vitorgalvao** [Homebrew/homebrew-cask#54811]
Followup to https://github.com/Homebrew/homebrew-cask/pull/54715.

These casks have an `appcast` but are using `:no_check`. They should be updated to have a proper `version` and `sha256`, even if the `url` is unversioned.

If you update some based on this issue, please comment below with a link to your PR, so we can update the list.

- [x] [amazon-workdocs](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/amazon-workdocs.rb)
- [x] [amazon-workspaces](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/amazon-workspaces.rb)
- [x] [anylist](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/anylist.rb)
- [x] [between](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/between.rb)
- [x] [bill](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/bill.rb)
- [x] [bluesense](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/bluesense.rb)
- [x] [cloudmounter](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/cloudmounter.rb)
- [x] [curio](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/curio.rb)
- [x] [dashlane](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/dashlane.rb)
- [x] [dyn-updater](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/dyn-updater.rb)
- [x] [easycrop](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/easycrop.rb)
- [x] [elmedia-player](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/elmedia-player.rb)
- [x] [eudic](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/eudic.rb)
- [x] [flirc](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/flirc.rb)
- [x] [fman](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/fman.rb)
- [x] [gotiengviet](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/gotiengviet.rb)
- [x] [haptic-touch-bar](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/haptic-touch-bar.rb)
- [x] [hubstaff](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/hubstaff.rb)
- [x] [ibackup-viewer](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/ibackup-viewer.rb)
- [x] [iholdem-indicator](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/iholdem-indicator.rb)
- [x] [ilok-license-manager](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/ilok-license-manager.rb)
- [x] [ios-console](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/ios-console.rb)
- [x] [ishowu-instant](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/ishowu-instant.rb)
- [x] [ishowu-studio](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/ishowu-studio.rb)
- [x] [izip](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/izip.rb)
- [x] [jumpshare](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/jumpshare.rb)
- [x] [lingo](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/lingo.rb)
- [x] [live-interior-3d-pro](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/live-interior-3d-pro.rb)
- [x] [livechat](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/livechat.rb)
- [x] [loopback](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/loopback.rb)
- [x] [macpilot](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/macpilot.rb)
- [x] [menubar-stats](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/menubar-stats.rb)
- [ ] [merlin-project](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/merlin-project.rb)
- [x] [mollyguard](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/mollyguard.rb)
- [x] [notebooks](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/notebooks.rb)
- [ ] [origami-studio](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/origami-studio.rb)
- [x] [path-finder](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/path-finder.rb)
- [x] [picka](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/picka.rb)
- [x] [product-hunt](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/product-hunt.rb)
- [x] [rest](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/rest.rb)
- [x] [sandvox](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/sandvox.rb)
- [ ] [scenery](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/scenery.rb)
- [x] [securedownloadmanager](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/securedownloadmanager.rb)
- [x] [the-clock](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/the-clock.rb)
- [x] [timing](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/timing.rb)
- [x] [tiny](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/tiny.rb)
- [x] [touchswitcher](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/touchswitcher.rb)
- [ ] [triumph](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/triumph.rb)
- [x] [tuneskit-m4v-converter](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/tuneskit-m4v-converter.rb)
- [ ] [twindocs](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/twindocs.rb)
- [x] [upthere-home](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/upthere-home.rb)
- [x] [watts](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/watts.rb)
- [x] [zenmate-vpn](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/zenmate-vpn.rb)

**Comment by vitorgalvao** [comment]
@ran-dall Yes, the way you’re doing it helps. Even if I need to edit them, having the correct `version` and `sha256` already saves a bunch of time. You can carry on. Thank you!

Out of curiosity, how did you find this issue?

**Comment by vitorgalvao** [comment]
> I found the issue just by looking at the issues board. Saw something I could help with and decided to do it.

Did the `good first issue` label help in making your decision to check the issue? I ask because I added it and also tweeted about the issue, so I’m curious if any of those things made a difference.

> I dare say that Cask-Repair makes it interesting, even a bit fun

Thank you. I’ve seen other people express similar sentiments in the past. That makes me glad to know!

**Comment by ran-dall** [comment]
@vitorgalvao I did find it curious. I don't know if it made a huge difference, but it did make me want to read into it.

My cask-repair went haywire this morning. I'm back at it now. However, in the process, I guess I updated 'cloudmaster'? I don't know if you could check it, but it seemed like it's updated somewhere along the process.

I don't know what really happened really happened with cask-repair. I'm thinking maybe my SSH-Key into GitHub Hub must have dislocated itself. It kinda fixed itself after I followed [this guide](https://github.com/jirsbek/SSH-keys-in-macOS-Sierra-keychain), and I read over some of the issues in your little-scripts repo. Basically, I reset everything and it started working again. Specifically, it was giving this error 
```origin is pointing to an incorrect remote (https://github.com/Homebrew/brew.git). Its beginning must match (https://|git@)github.com[/:]Homebrew/homebrew-cask.``` 
so I fixed it after cleaning everything and 
```git remote set-url origin https://github.com/Homebrew/homebrew-cask```

Anyways, I'm back at it now...

@chrisfinazzo Thanks for picking up those up.

**Comment by ran-dall** [comment]
@vitorgalvao NM. It was still messed up. I fixed it. Lol.

Here everything to the the first G.

Here's the new ones:
https://github.com/Homebrew/homebrew-cask/pull/54908 - cloudmounter
https://github.com/Homebrew/homebrew-cask/pull/54909 - dashlane
https://github.com/Homebrew/homebrew-cask/pull/54910 - dyn-updater
https://github.com/Homebrew/homebrew-cask/pull/54911 - easycorp
https://github.com/Homebrew/homebrew-cask/pull/54912 - elmedia-player
https://github.com/Homebrew/homebrew-cask/pull/54913 - eudic
https://github.com/Homebrew/homebrew-cask/pull/54914 - flirc
https://github.com/Homebrew/homebrew-cask/pull/54915 - gotiengviet

**Comment by ran-dall** [comment]
https://github.com/Homebrew/homebrew-cask/pull/54916 - haptic-touch-bar
https://github.com/Homebrew/homebrew-cask/pull/54917 - hubstaff
https://github.com/Homebrew/homebrew-cask/pull/54918 - ibackup-viewer
https://github.com/Homebrew/homebrew-cask/pull/54919 - iholdem-indicator
https://github.com/Homebrew/homebrew-cask/pull/54921 - ilok-license-manager (might have made an error in the GM)
https://github.com/Homebrew/homebrew-cask/pull/54920 - ios-console
https://github.com/Homebrew/homebrew-cask/pull/54922 - ishowu-instant
https://github.com/Homebrew/homebrew-cask/pull/54923 - ishowu-studio

**Comment by ran-dall** [comment]
@vitorgalvao Here's the rest. Let me know if you want me to redo the ones that I messed up...

https://github.com/Homebrew/homebrew-cask/pull/54940 - timing
https://github.com/Homebrew/homebrew-cask/pull/54939 - tiny
https://github.com/Homebrew/homebrew-cask/pull/54942 - touchswitcher
https://github.com/Homebrew/homebrew-cask/pull/54941 - triumph
https://github.com/Homebrew/homebrew-cask/pull/54943 - tuneskit-m4v-converter
https://github.com/Homebrew/homebrew-cask/pull/54944 - twindocs
https://github.com/Homebrew/homebrew-cask/pull/54945 - upthere-home
https://github.com/Homebrew/homebrew-cask/pull/54946 - watts
https://github.com/Homebrew/homebrew-cask/pull/54947 - zenmate-vpn

**Comment by ran-dall** [comment]
@vitorgalvao triumph was merged (https://github.com/Homebrew/homebrew-cask/pull/54941), and Twindocs is still open (https://github.com/Homebrew/homebrew-cask/pull/54944). Scenery no longer exists.

**Comment by vitorgalvao** [comment]
@ran-dall Thank you for the good work!

**Comment by core-code** [comment]
thanks, i think its awesome how @ran-dall stepped in here.

another thing where the community could help and which would be a  great "good first issue" in my opinion is our work we have been doing on pr https://github.com/Homebrew/homebrew-cask/issues/52837

basically we are automatically identifying casks that are very likely to be outdated and which should be cask-repaired. the idea here was to automate that too, but that hasn't played out yet, and since there are always around 20% false positives, i'm not sure how to do that (unless you blindly submit anything thats on the server - which is a bad idea).

each day the new file is stored here:
http://macupdater.net/cask-repair/
most interesting is basically the 'processed' file, i.e. the newest one ATM is:
http://macupdater.net/cask-repair/cr-processed-2018-11-17.txt

me and mostly @suschizu have been going over this list each day but it would be awesome if the community could step in here as we won't have much time for that for some time.

basically whats needed is to go over the list and for each app:
• have a look if the proposed new version actually exists (i.e. on their homapage), and if it does, do a cask-repair
• if it doesn't exist (=>false positive) let me know so i can put it on the ignore list (there are 3 different ones as explained in the thread) so it won't show up again. basically i have to update the postprocessing script ( http://macupdater.net/cask-repair/postprocessCaskRepairOutput.py.txt )

so, @vitorgalvao it would be awesome if you could create another 'issue' like that with the 'good first issue' label with the info i've written above. if you don't think its a good idea, or e.g. want me to create the issue just let me know.

**Comment by ran-dall** [comment]
@core-code Thanks for your recognition. Couldn't have been done without the help of the great @vitorgalvao. 

I'd be happy to keep track of these files and keep them up to date. I've already been helping people out with the stuff that @vitorgalvao has been schooling me on. I think I've come a long way. It's amazing how many people don't read the `/docs`or the PR / Issue template. Anyways, **I'll get the outdated Casks from today updated shortly.** 

I'm usually glued to my computer 34/7/365 (for a variety of professional, business, and enthusiast reasons, much to my wife's dismay), so this should be something I can generally be depended on to take care of for you. If for whatever reason I'll be gone (and without a computer), I'll let you guys know.

Besides, I enjoy homebrew-cask; and `cask-repair`is pretty awesome too (much credit and respect to @vitorgalvao ).

I've also read through #52837. It gave me a lot of insight. Thanks for sharing. I found @brianmorton automation script a few days ago and was playing around with it. I didn't realise the whole picture as much as I do know. I have a few ideas I've been playing around the last few days that I might add to the thread (#52837) if they prove to reliable. 

Side-note: The raw data output you're doing is excellent stuff. Good work!

**Comment by ran-dall** [comment]
@core-code `gog-galaxy`, `goland`, `wacom-tablet`, and `wondershare-video-converter-ultimate` need closer looks.

**Comment by vitorgalvao** [comment]
> or e.g. want me to create the issue just let me know.

Please do. The system is your creation so you’ll be explain it better. I’ll add the tag.

**Comment by core-code** [comment]
@ran-dall  awesome, thanks!

gog-galaxy: think its a false positive, i've made sure this version is ignored. if this happens more often, we'll need to take more drastic action here
wacom-tablet: i already did a PR here to update it (it lives in the homebrew-cask-drivers repository, as the '=' prefix indicates). this may already have gone through by the time you looked at it. my system currently can't filter things where there is already a pull-request open
wondershare-video-converter-ultimate: this was a valid update and i've submitted a PR. the wondershare guys update their files in-place most of the time
goland: sorry will have to take a look at this later on

new output for today is up.

p.s. i also feel like i am glued to the computer 34 hours per day ;)

**Comment by core-code** [comment]
> Please do. The system is your creation so you’ll be explain it better. I’ll add the tag.

thanks, i'll do so later on

**Comment by ran-dall** [comment]
@core-code NP. I was just getting my homework started. Now, my homework is done for the day, sir!

Notes for the day:
cmake 3.13.0 - is a release candidate
datasciencestudio - is already open
intellij-idea and goland - it's strange because on their website they have the same version as HBC, but those versions do exist I'm thinking maybe it's pre-release or something of the sort.

Everything else is updated.

Side Note: Maybe we should move this to the new issue?

**Comment by core-code** [comment]
sorry, off on an (even more ;) remote island and finding little time for proper things ATM. i try to run the script daily though.

i've made sure the cmake release candidate is ignored now. 

intellij-idea and goland .. .quite sure those vendors distribute beta versions so it might be better to just ignore those. will need to look if they have other ways to keep them updated them though.

**Comment by ran-dall** [comment]
@core-code I was wondering what happened to you! I totally understand. I was looking for the output this morning and didn't see it. Anyways, jobs done...

**double-commander** - False positive (https://sourceforge.net/p/doublecmd/wiki/Download/)
**data-dog** - **STILL OPEN** (https://github.com/Homebrew/homebrew-cask/pull/54739)
**knockknock** - BETA (https://bitbucket.org/objective-see/deploy/downloads/)
**progressive-downloader** - False positive (https://www.macpsd.net)

**Comment by ran-dall** [comment]
@core-code Are false positives apps that auto-update maybe?

**Comment by core-code** [comment]
thanks. i've updated the false positive list and uploaded new output.

i don't think there is a connection between "auto-updates" and 'false positive'.

'false positive' here just means that one of our users is using a version of an app that is not an official release - e.g. beta or test versions. there is probably no connection between whether an app can automatically update and whether an app distributes beta versions.

**Comment by ran-dall** [comment]
@core-code Nice! I'll get to it shortly. I'll let you know when I'm done.

I was just critically thinking, and the users are getting these updates somewhere. I was investigating with JetBrains (as I was given a free Lifetime license for winning an AI hackathon a few months ago) and it seems that these versions are installed via JetBrains Toolbox. So it kinda acts like that of an auto-update as JetBrains seems to be pushing out updates almost every week thru JetBrains Toolbox. Which got me to thinking, these apps basically update on their own if you have the apps managed via JetBrains Toolbox; which was the basis of my comment.

**Comment by ran-dall** [comment]
@core-code Here's the work...

**datadog-agent** - **STILL OPEN (#54739 )**
**alfred** - false positive (https://www.alfredapp.com/)
**texstudio** - pre-release (https://github.com/texstudio-org/texstudio/releases/) (also ` conflicts_with cask: 'texstudio-beta'`)

**Comment by ran-dall** [comment]
@core-code Good work on #55268 ! 👍

**Comment by ran-dall** [comment]
@brianmorton I really appreciate that! I will keep you in the loop. Thanks for all your great work too!

**Comment by core-code** [comment]
> @core-code Good work on #55268 ! 👍

thanks, good job on the cask-repairs ;)

**Comment by ran-dall** [comment]
@core-code Couldn't be done with your output. 😁

One small request though, if you could leave me a mention when you put up new outputs. That would be helpful.

**Comment by core-code** [comment]
i'll try to leave a note in #55268 from now on

**Comment by ran-dall** [comment]
@core-code Thank You!

**Comment by core-code** [comment]
i think the intellij (idea, pycharm, goland) updates that were previously false-positive are out in final now

**Comment by core-code** [comment]
ah thanks good to know that bot exists
@leipert many thanks for jcbot. i'll be excluding these casks from my system as they are already handled perfectly.

**Comment by leipert** [comment]
@core-code Thanks for the kind words 😄
